src/peerbook/libp2p_peerbook.erl

```diff
@@ -562,3 +563,27 @@ install_gossip_handler(TID, Handle) ->
             libp2p_group_gossip:add_handler(G,  ?GOSSIP_GROUP_KEY, {?MODULE, Handle}),
             G
     end.
+
+get_signed_metadata(State = #state{tid=TID, metadata_ref=MR}) ->
+    case MR of
+        undefined ->
+            MetaDataFun = libp2p_config:get_opt(libp2p_swarm:opts(TID), [libp2p_peerbook, signed_metadata_fun],
+                                                fun() -> #{} end),
+
+            Parent = self(),
+            {_, Ref} = spawn_monitor(fun() ->
+                                             %% if the metadata fun crashes, use the old metadata
+                                             try MetaDataFun() of
```

Comment:
So maybe remove the try catch and let spawn_monitor do it's job to leave the old metadata alone?

Reply:
or store the spawned pid in the state if the spawn is not going to crash because you try catch everything and send an error back to the parent so it can clear the stored pid

Reply:
I.e. ```-spec get_signed_metadata(#state{}) -> #state{}.
get_signed_metadata(State=#state{metadata_fun=MetaDataFun}) ->
    case State#state.metadata_ref of
        undefined ->
            Parent = self(),
            {_, Ref} = spawn_monitor(fun() ->
                                             Parent ! {signed_metadata, MetaDataFun()}
                                     end),
            State#state{metadata_ref = Ref};
        _ ->
            State
    end.
```

Reply:
Right, this is a holdover from the old code.